src/renderers/shared/ReactDebugTool.js

```diff
@@ -68,7 +69,7 @@ var ReactDebugTool = ((null: any): typeof ReactDebugTool);
 if (__DEV__) {
   const hooks = [];
   const didHookThrowForEvent = {};
-  const ReactComponentTreeHook = require('react/lib/ReactComponentTreeHook');
+  const ReactComponentTreeHook: ComponentTreeHookDevType = (require('react/lib/ReactComponentTreeHook'): any);
```

Comment:
@spicyj @bvaughn @gaearon 

I've updated the PR with some guards. Let me know if there's something I may have missed? I tested locally with React (min) and ReactDOM (non-min) and it's working, although that's not going to test all the code paths that dev code may go through – so please let me know if something doesn't look right.